### DISCUSSION TITLE
What is the difference between `cute::Tensor::element_type` and `cute::Tensor::value_type`?

### DISCUSSION THREAD

**Discussion by hyhieu** [NVIDIA/cutlass#1409] [Q&A] [answered]
The question is specifically about this code snippet:

https://github.com/NVIDIA/cutlass/blob/ffa34e70756b0bc744e1dfcc115b5a991a68f132/include/cute/tensor.hpp#L110-L113

But I do also see `element_type` and `value_type` coming up at other contexts in `CuTe` and `CUTLASS`. Is there a general guidelines what they mean?

Thanks!

**Comment by hwu36** [comment]
@thakkarV

**Reply by thakkarV** [reply, marked as answer]
Usually they are the same but there are cases where they need not be. Value type is the logical value the tensor stores. Element type is the underlying physical storage for it.
